lib/traffic_source_parser/result/utmz.rb

```diff
@@ -6,6 +6,18 @@ class Utmz < TrafficSourceParser::Result::Generic
 
       def initialize(utmz_hash)
         super(utmz_hash)
+
+        self[:source] = get_known_source(utmz_hash) if use_known_sources?(utmz_hash)
+      end
+
+      private
+
+      def use_known_sources?(utmz_hash)
+        TrafficSourceParser::Parser::ReferrerParser.referrers_list.has_key?(utmz_hash[:source])
+      end
+
+      def get_known_source(utmz_hash)
+        TrafficSourceParser::Parser::ReferrerParser.referrers_list[utmz_hash[:source]]['source']
```

Comment:
Line is too long. [96/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.